src/styles.ts

```diff
@@ -6,48 +6,90 @@ import * as sass from "node-sass";
 import * as postcss from "postcss";
 import * as styleLint from "stylelint";
 import * as log4js from "log4js";
+import * as sassGraph from "sass-graph";
+import * as chokidar from "chokidar";
 
 import {StylesCompileConfig, StylesLintingConfig} from "./config/config.interface";
 import utils from "./utils";
 
 export class StyleCompiler extends EventEmitter {
 
-  private logger = log4js.getLogger("global-styles");
+  static logger = log4js.getLogger("compile-styles");
+  private logger: log4js.Logger = StyleCompiler.logger;
+  private watcher: fs.FSWatcher;
+  private entryWatchers: {[file: string]: fs.FSWatcher} = {};
 
   constructor(private config: StylesCompileConfig) {
     super();
   }
 
   start() {
-    this.run();
-    if (this.config.watchPattern) {
-      utils.watch(this.config.watchPattern as string, () => this.run());
+    if (this.config.watch) {
+      this.watcher = chokidar.watch(this.config.entry as string[], _.pick(this.config, ["cwd"]))
+        .on("unlink", (entry: string) => {
+          this.entryWatchers[entry].close();
+          this.emit("unwatch", entry);
+        })
+        .on("add", (entry: string) => {
+          this.watchForChanges(entry);
+          this.run(entry);
+          this.emit("watch", entry);
+        });
+    } else {
+      this.run(this.config.entry);
     }
 
     return this;
   }
 
-  run() {
-    this.compile()
-      .then(
-        () => {
-          this.logger.debug(`Global styles written to ${this.config.output}`);
-          this.emit("success");
-        },
-        (err) => {
-          this.logger.error("Error processing global styles:", err);
-          this.emit("error", err);
+  stop() {
+    if (this.watcher) {
+      this.watcher.close();
+    }
+    _.each(this.entryWatchers, watcher => watcher.close());
+  }
+
+  private watchForChanges(entry: string) {
+    let filesToWatch = this.getFilesToWatch(entry);
+    const watcher = utils.watch(filesToWatch, () => {
+      this.run(entry);
+      filesToWatch.forEach((file: string) => watcher.unwatch(file));
+      filesToWatch = this.getFilesToWatch(entry);
+      filesToWatch.forEach((file: string) => watcher.add(file));
+    }, _.pick(this.config, ["cwd"]));
+    this.entryWatchers[entry] = watcher;
+  }
+
+  private getFilesToWatch(entry: string) {
+    return _.keys(sassGraph.parseFile(`${this.cwd()}${entry}`, {
```

Comment:
Please use `path.join` instead of relying on the trailing `/` - ensures cross-platform compatibility.